### ISSUE TITLE
Decide which public interface we want

### ISSUE THREAD

**Issue by gonzus** [gonzus/http-nghttp2#7]
Right now we have two Perl modules:
1. NGHTTP2::Session: it simply loads the XS wrapper
2. NGHTTP2::Client: it implements our client (and uses NGHTTP2::Session)

And we have two test programs:
1. tryit.pl: loads NGHTTP2::Session directly, a lower-level implementation
2. run.pl: loads NGHTTP2::Client (and NGHTTP2::Request, but that is a pure Perl helper object), a higher-level implementation

It seems advisable to clean this up a bit. Maybe it is OK for NGHTTP2::Session to stay, but if we agree this should not be visible to client code, then we should stick to NGHTTP2::Client as our public API (and maybe get rid of the tryit.pl example, or convert that into a test, or something).

**Comment by xsawyerx** [comment]
I agree we should clean it up. I have written some tests and improved them, but I would like to write more. Eventually these files should dissipate between the tests and the documentation.
